resolwe/flow/managers/base.py

```diff
@@ -74,26 +168,365 @@ def discover_engines(self):
+        os.mkdir(output_path, mode=dir_mode)
+        # os.mkdir is not guaranteed to set the given mode
+        os.chmod(output_path, dir_mode)
+        return output_path
+
+    def _prepare_context(self, data_id, data_dir, runtime_dir, **kwargs):
+        """Prepare settings and constants jsons for the executor.
+
+        Settings and constants provided by other ``resolwe`` modules and
+        :class:`~django.conf.settings` are all inaccessible in the
+        executor once it is deployed, so they need to be serialized into
+        the runtime directory.
+
+        :param data_id: The data object id being prepared for.
+        :param data_dir: The target execution directory for this data
+            object.
+        :param runtime_dir: The target runtime support directory for
+            this data object; this is where the environment is
+            serialized into.
+        :param kwargs: Extra settings to include in the main settings
+            file.
+        """
+        files = {}
+
+        settings_dict = dict(kwargs)
+        settings_dict.update(self.settings_actual)
+        settings_dict['DATA_DIR'] = data_dir
+        settings_dict['REDIS_CHANNEL_PAIR'] = state.MANAGER_EXECUTOR_CHANNELS
+        files[ExecutorFiles.SETTINGS] = settings_dict
+
+        # add scheduling classes
+        files[ExecutorFiles.PROCESS_META] = {
+            k: getattr(Process, k) for k in dir(Process)
+            if k.startswith('SCHEDULING_CLASS_') and isinstance(getattr(Process, k), str)
+        }
+
+        # add Data status constants
+        files[ExecutorFiles.DATA_META] = {
+            k: getattr(Data, k) for k in dir(Data)
+            if k.startswith('STATUS_') and isinstance(getattr(Data, k), str)
+        }
+
+        # extend the settings with whatever the executor wants
+        self.executor.extend_settings(data_id, files)
+
+        # save the settings into the various files in the data dir
+        settings_dict[ExecutorFiles.FILE_LIST_KEY] = list(files.keys())
+        for file_name in files:
+            file_path = os.path.join(runtime_dir, file_name)
+            with open(file_path, 'wt') as json_file:
+                json.dump(files[file_name], json_file, cls=SettingsJSONifier)
+
+    def _prepare_executor(self, data_id, executor):
+        """Copy executor sources into the destination directory.
+
+        :param data_id: The data object id being prepared for.
+        :param executor: The fully qualified name of the executor that
+            is to be used for this data object.
+        :return: Tuple containing the relative fully qualified name of
+            the executor class ('relative' to how the executor will be
+            run) and the path to the directory where the executor will
+            be deployed.
+        :rtype: (str, str)
+        """
+        # Both of these imports are here only to get the packages' paths.
+        import resolwe.flow.executors as executor_module
```

Comment:
@jberci, did you see my comment that `executors` are a package so it would be better if you renamed `executor_module` to `executors_package`?